### ISSUE TITLE
The termguicolors option must be set

### ISSUE THREAD

**Issue by macguirerintoul** [bluz71/vim-nightfly-guicolors#6]
Hi there,

I love the theme, but can't seem to get it to work. I'm not sure what the problem is, I've tried following articles and suggestions from various sources. The issue is when I start vim, it prints out  the following and the colour scheme does not change:

```
The termguicolors option must be set
Press ENTER or type command to continue
```

I'm using iTerm2 build 3.3.9 and vim version 8.1.1312. My iTerm report terminal type is set to `xterm-256color`. My OS is macOS Catalina 10.15.3.

My `.vimrc`: https://github.com/macguirerintoul/dotfiles/blob/master/.vimrc

```
" Plug setup
call plug#begin()
Plug 'sheerun/vim-polyglot'
Plug '/usr/local/opt/fzf'
Plug 'junegunn/fzf.vim'
Plug 'neoclide/coc.nvim', {'branch': 'release'}
Plug 'preservim/nerdtree'
Plug 'jiangmiao/auto-pairs'
Plug 'dense-analysis/ale'
Plug 'bluz71/vim-nightfly-guicolors'
call plug#end()

" syntax enable
set background=dark

" set Vim-specific sequences for RGB colors
set termguicolors
let &t_8f = "\[38;2;%lu;%lu;%lum"
let &t_8b = "\[48;2;%lu;%lu;%lum"

colorscheme nightfly

" Configure NERDTree
let NERDTreeShowHidden=1

" Configure ALE
let g:ale_fixers = {'javascript': ['eslint'], 'scss': ['stylelint']}
let g:ale_fix_on_save = 1
let g:ale_lint_on_text_changed = 'never'
let g:ale_lint_on_insert_leave = 0

" Ctrl + n toggles NERDTree
map <C-n> :NERDTreeToggle<CR>
" Ctrl + p opens fzf in git files mode
map <C-p> :GFiles<CR>
set tabstop=2 " set display width of tab
set softtabstop=1
set cursorline
set wildmenu
set showmatch
" turn hybrid line numbers on
:set number relativenumber
:set nu rnu
```

Any suggestions?

**Comment by bluz71** [comment]
Add "set termguicolors" in your vimrc file.

**Comment by macguirerintoul** [comment]
Hi @bluz71, it's there on line 17. Directly under `" set Vim-specific sequences for RGB colors`

**Comment by bluz71** [comment]
Sorry I did not see termguicolors in the middle of your vimrc. My bad

Let me stew on this.

**Comment by bluz71** [comment]
Please move:

```viml
set termguicolors
```

to the very top of your `.vimrc` file and try again. It needs to be set before you load the `nightfly` plugin.

Let me know if that fixes it or not.

**Comment by macguirerintoul** [comment]
Unfortunately that did not fix it.

**Comment by bluz71** [comment]
Ok, drastic action needed. Please cut your `~/.vimrc` back to this minimal configuration:

```viml
set termguicolors

call plug#begin()
Plug 'bluz71/vim-nightfly-guicolors'
call plug#end()

colorscheme nightfly
```

Does that work?

**Comment by macguirerintoul** [comment]
Unfortunately not 😕 I still get the following message:
```
The termguicolors option must be set
Press ENTER or type command to continue
```

**Comment by bluz71** [comment]
Something very strange is happening. I don't understand why the error message is being emitted. I am testing with 8.2 in iTerm 3.3.9 on Catalina and nightfly works just fine.

What happens with this `~/.vimrc` configuration:

```viml
set termguicolors

if !has("gui_running") && !exists('&termguicolors')
    echomsg "A modern version of Vim with termguicolors is required"
    finish
elseif !has("gui_running") && !&termguicolors
    echomsg "The termguicolors option must be set"
    finish
endif
```

Also, what happens if you use Neovim instead of Vim?
For seamless Neovim usage with your existing Vim setup please create the file `~/.config/nvim/init.vim` with the content:

```viml
set runtimepath^=~/.vim runtimepath+=~/.vim/after
let &packpath = &runtimepath
source ~/.vimrc
```

Lastly, please test with a version of Vim 8.2 instead of 8.1.1312.

**Comment by macguirerintoul** [comment]
Hmm. So using Vim 8.1.1312 (default I had on my Mac) and the latest `.vimrc` you provided, it still prints `The termguicolors option must be set`. However, I ran `brew install vim` and upgraded to 8.2.0400 and my original `.vimrc` works beautifully. Thanks for all your help!

**Comment by bluz71** [comment]
Strange one. It seems like your Vim 8.1.1312 may have been slightly broken?

Anyway, problem solved, and that's always a good thing.

Stay safe.

**Comment by esquishesque** [comment]
fwiw, a similar thing happened to me, with vim 8.0 on macos, termguicolors was definitely working but not with your plugin and it gave that error message (it actually gave the message twice every time).  i reinstalled vim via homebrew and now it works, with no further changes to my .vimrc.  thanks for the colorscheme, it's lovely!

**Comment by bluz71** [comment]
I just test the macOS Big Sur native version of Vim ......AND it is super weird.

It claims to be Vim 8.2 but the `termguicolors` option appears to be a non-operation. I tried the [nord](https://github.com/arcticicestudio/nord-vim) theme and it is broken with the macOS Vim.

If I use Homebrew Vim or standard Neovim everything just works.

I think Apple stripped out `termguicolors` back-end, or some such. Whatever happened, it is definitely broken with respect to `termguicolors`.

With that knowledge, I have updated the startup error for the nightfly theme to:

```
The termguicolors option must be set. 
Be aware macOS default Vim is broken, use Homebrew Vim instead.
```

Thanks for letting me know, this little update should help future macOS users of this theme.

Cheers.
